core/http_api.php

```diff
@@ -137,9 +137,9 @@ function http_security_headers() {
 		$t_avatar_img_allow = '';
 		if ( config_get_global( 'show_avatar' ) ) {
 			if ( http_is_protocol_https() ) {
-				$t_avatar_img_allow = "; img-src 'self' https://secure.gravatar.com:443";
+				$t_avatar_img_allow = "; img-src 'self' ".config_get( 'show_avatar_url_secure' ).":443";
 			} else {
-				$t_avatar_img_allow = "; img-src 'self' http://www.gravatar.com:80";
+				$t_avatar_img_allow = "; img-src 'self' "config_get( 'show_avatar_url' )":80";
```

Comment:
Are you missing concat dots here?  Also in general there should be spaces around the dot, you please fix the one for the secure case.